datatypes/src/primitives/time_interval.rs

```diff
@@ -212,6 +215,8 @@ impl TimeInterval {
         other == self
             || value_in_range(self.start, other.start, other.end)
             || value_in_range(other.start, self.start, self.end)
+            || { value_in_range_inclusive(self.start, other.start, other.end) && self.is_instant() }
```

Comment:
- why do you use curly braces? please use regular braces here
- I don't think it's correct to use `inclusive here`. The end of a time interval is exclusive
- can you please add tests that would have failed before your change and do work now

Reply:
The inclusive check is only for `TimeInstance`, since `value_in_range` returns `false` when the other `TimeInstance` is at the end of the `TimeInterval`.